internal/rbd/rbd_util.go

```diff
@@ -707,25 +727,17 @@ func genVolFromVolumeOptions(ctx context.Context, volOptions, credentials map[st
 	// if no image features is provided, it results in empty string
 	// which disable all RBD image features as we expected
 
-	imageFeatures, found := volOptions["imageFeatures"]
-	if found {
-		arr := strings.Split(imageFeatures, ",")
-		for _, f := range arr {
-			if !supportedFeatures.Has(f) {
-				return nil, fmt.Errorf("invalid feature %q for volume csi-rbdplugin, supported"+
-					" features are: %v", f, supportedFeatures)
-			}
-		}
-		rbdVol.imageFeatureSet = librbd.FeatureSetFromNames(arr)
+	if rbdVol.Mounter, ok = volOptions["mounter"]; !ok {
+		rbdVol.Mounter = rbdDefaultMounter
+	}
+	if err = rbdVol.validateImageFeatures(volOptions["imageFeatures"]); err != nil {
+		klog.Errorf(util.Log(ctx, "failed to validate image features %v"), err)
+		return nil, err
 	}
 
-	util.ExtendedLog(ctx, "setting disableInUseChecks on rbd volume to: %v", disableInUseChecks)
 	rbdVol.DisableInUseChecks = disableInUseChecks
 
-	rbdVol.Mounter, ok = volOptions["mounter"]
-	if !ok {
-		rbdVol.Mounter = rbdDefaultMounter
-	}
+	util.ExtendedLog(ctx, "setting disableInUseChecks: %t imagefeatures: %v mounter: %s", disableInUseChecks, rbdVol.imageFeatureSet, rbdVol.Mounter)
```

Comment:
I think you need to log `rbdVol.imageFeatureSet.Names()` to get some useful results.

Reply:
done